File: Backend/NaissusEvents/Models/Reservation.cs

using System.ComponentModel.DataAnnotations;


namespace Models
{

    public class Reservation
    {
        [Key]
        public int Id { get; set; }

        public virtual Event Event { get; set; }


        public Table Table {get; set;}
        
        public virtual AppUser MyUser {get;set;}

    }
}

File: Frontend/src/app/models/Reservation.ts
import { myEvent } from "./myEvent";
import { User } from "./User";
import { Table } from "./Table";

export class Reservation{

    id: number;
    event: myEvent;
    table: Table;
    myUser: User;

    constructor(id: number, eventId: myEvent, tableId: Table, userId: User) {
        this.id = id;
        this.event = eventId;
        this.table = tableId;
        this.myUser = userId;
    }


}